### DISCUSSION TITLE
How to uninstall it properly

### DISCUSSION THREAD

**Discussion by sergeevabc** [vkbo/novelWriter#739] [General]
Dear Veronica,
I see `setup_windows.bat` which does a lot, but there is no `uninstall.bat` or something like that.
Could you be so kind to add it, or at least provide me (us, the community) a step-by-step guide?

**Comment by vkbo** [comment]
The install on windows is nothing more than creating the start menu and desktop icons. You can just delete them again.

It also adds a few entries into the registry, but those can be ignored.

**Reply by sergeevabc** [reply]
Nevertheless, Veronica, consider making some kind of uninstaller, because unnecessary leftovers such as Python QT-packages and registry entries can be called a contribution to the digital pollution. Make your app is a decent guest, not a bully.

**Reply by vkbo** [reply]
The dependencies are installed via pip, and pip doesn't have an option to determine which packages installed via recursive dependency are no longer needed. The documentation lists which direct dependencies novelWriter has, but uninstalling those will still leave a lot of packages on the system. There isn't much I can do about that. pip isn't a very sophisticated package manager.

**Reply by vkbo** [reply]
I added a cleanup option for XDG desktops a few days ago. I plan to add one for Windows too that will remove icons and registry keys automatically. I don't currently have access to a Windows computer to write the code and test it.

The pip install step is on pip. If I just brute force remove pip packages, I may break the package dependencies for other packages if the user uses pip for other things than novelWriter. PyPi needs to fix their package manager. I could make a mention of it in the documentation though.

**Reply by sergeevabc** [reply]
> The pip install step is on pip. If I just brute force remove pip packages,

Or, perhaps, some kind of a force switch with warning for advanced users. I am willing to take the risk.

**Comment by vkbo** [comment]
I've added a `win-uninstall` command to `setup.py` that deletes icons and registry keys. See PR #749.

I also added a `.bat` file as you requested that will call this command, and also run the `pip uninstall -r requirements.txt` command. It will remove the direct dependencies, but dependencies of dependencies I cannot do much about until the `pip` potentially adds such a feature.

Edit: I see in hindsight that I shouldn't have converted this from an issue to a discussion, but I initially interpreted this as a question. I created an issue again that points back here for the record. #743.

**Reply by sergeevabc** [reply]
Thank you for the diligent attitude towards the issue, Veronica.
